activejob/lib/active_job/arguments.rb

```diff
@@ -76,9 +107,14 @@ def serialize_argument(argument)
         when ActiveSupport::HashWithIndifferentAccess
           serialize_indifferent_hash(argument)
         when Hash
-          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s)
+          symbol_keys = argument.each_key.grep(Symbol).map!(&:to_s)
+          aj_hash_key = if Hash.ruby2_keywords_hash?(argument)
+            RUBY2_KEYWORDS_KEY
```

Comment:
The difference from #38260 is storing only extra 3 bytes in exchange for requiring deployment order (deploy this change first, then update Ruby 2.6 to 2.7).

This is just a thought experiment, safe (#38260) vs compact (#38267).

Reply:
This is a great idea. I prefer this one.